City-of-Helsinki/drupal-hdbt
# UHF-9441: Fetch timeout

# [UHF-9441](https://helsinkisolutionoffice.atlassian.net/browse/UHF-9441)
Set a timeout for react app searches to prevent 'infinite' loading visuals.

## What was done
- Added new `useTimeoutFetch` wrapper to timeout requests
- Added use of timeout fetch where necessary
- Made school search behave better with timeout requests
- Made no results error more visible

## How to install

This change affects all the react search apps. You may want to test a couple of them. You can find the search apps in these local urls:

Etusivu:
- https://helfi-etusivu.docker.so/uutiset/etsi-uutisia

Rekry:
- https://helfi-rekry.docker.so/fi/avoimet-tyopaikat/etsi-avoimia-tyopaikkoja

SOTE:
- https://helfi-sote.docker.so/fi/sosiaali-ja-terveyspalvelut/lasten-ja-perheiden-palvelut/aitiys-ja-lastenneuvolat
- https://helfi-sote.docker.so/fi/sosiaali-ja-terveyspalvelut/terveydenhoito/terveysasemat

KASKO:
- https://helfi-kasko.docker.so/fi/kasvatus-ja-koulutus/perusopetus/peruskoulut

KYMP:
- https://helfi-kymp.docker.so/fi/kaupunkiymparisto-ja-liikenne/kaupunkisuunnittelu-ja-rakentaminen/suunnitelmat-ja-rakennushankkeet
- https://helfi-kymp.docker.so/fi/kaupunkiymparisto-ja-liikenne/kunnossapito/katujen-kunnossapito/katujen-talvikunnossapito

On any of you instance of choice, do:
* Make sure your instance is up and running on latest dev branch.
  * `git pull origin dev`
  * `make fresh`

...and make sure you have elastic data indexed: 
*  drush sapi-rt
* drush sapi-i
* drush cr

...and install this branc:
* Update the HDBT theme
  * `composer require drupal/hdbt:dev-UHF-9441-fetch-timeout`
* Run `make drush-cr`

## How to test

On any of the search apps, verify first that you have it up and working. Then, pause the elastic container. You can do this by using Docker Desktop. Go to containers -> Locate the elastic container under the project you are using -> Select Pause from the kebab menu on the right. 
When the elastic container is paused. Try to reload the page. You should see an error on the page within ~10-20 second depending on the search app. (Vs. it used to just show the loading spinner infinitely)

## Continuous documentation



## Translations





[UHF-218]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-218?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[UHF-9441]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-9441?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
